### ISSUE TITLE
Update logic for `swiper-slide-zoomed` or add different `swiper-slide-zooming` class

### ISSUE THREAD

**Issue by garrettmaring** [nolimits4web/swiper#2995]
The class `swiper-slide-zoomed` (default, this is configurable) is applied to a slide when a user double-taps on a slide. Currently, this class is **not** added when the user uses the "pinch" gesture. It would be helpful to be able to share logic across a slide being zoomed in either of the two fashions.

An initial pass could add the `swiper-slide-zoomed` class to a slide no matter how it was zoomed. A possible expansion of this feature could be to add a `swiper-slide-zooming` class while a zoom is occuring.

**Comment by garrettmaring** [comment]
Just FYI @nolimits4web I still think this is a good issue to tackle! Stale bot is a grump.

**Comment by ncla** [comment]
This still is an issue.

The documentation says the following:

> Swiper supports zoom images functionality (similar to what you see on iOS when browsing single photo) where you can zoom-in image by pinch gesture and or by zoom-in/out by double tap on it.

Going off from this text, a zoom should be considered to be true when either double tapping or pinch zooming has been used.

I created a simple reproduction by styling a zoomed slide (halved opacity) and I opened the repro site on iPhone iOS 15.1 Safari. Using latest Swiper (version 7.4.1).

When zooming a slide by pinching the screen, the `.swiper-slide-zoomed` CSS class is not added to the `.swiper-slide` element. Double tapping does however attach this class.

https://codesandbox.io/s/swiper-zoom-forked-476r4
